### PR TITLE
Update CODEOWNERS to avoid project files being owned by just one person

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

Only exception: CI files, as no-one apart from @mwu-tow is currently modifying them anyway

### Checklist

Please include the following checklist in your PR:

- [ ] The documentation has been updated if necessary.
- [ ] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -32,16 +31,13 @@ Cargo.toml
...
+/distribution/ @4e6 @jdunkerley @radeusgd
 /engine/ @4e6 @jaroslavtulach @hubertp
 /project/ @4e6 @jaroslavtulach @hubertp
 /test/ @jdunkerley @radeusgd
-/tools/ @4e6
+/tools/ @4e6 @jaroslavtulach @radeusgd
 
 # Enso Libraries
 # This section should be amended once the engine moves to /app/engine
 /distribution/lib/ @jdunkerley @radeusgd
```
Comment (radeusgd) [inline .github/CODEOWNERS:42]: I left this, as these are purely library files and so no sense in pinging @4e6 every time me and James modify a library file (almost every of our PRs).

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -32,16 +31,13 @@ Cargo.toml
...
 /std-bits/ @jdunkerley @radeusgd
-
-# Engine
-/app/engine/ @4e6
```
Comment (radeusgd) [inline .github/CODEOWNERS:47]: This directory currently does not exist, it shall be re-added once the repository is actually restructured.

**Review by JaroslavTulach — APPROVED** [review]
I happily accept being a new codeowner for `lib/scala` and `tools`.

File: .github/CODEOWNERS (hunk 2 above)
Comment (xvcgreg) [reply .github/CODEOWNERS:47]: Please create issue on pivotal for that.

**Review by xvcgreg — APPROVED** [review]

**Review by hubertp — APPROVED** [review]

File: .github/CODEOWNERS (hunk 2 above)
Comment (radeusgd) [reply .github/CODEOWNERS:47]: > Please create issue on pivotal for that.

OK: https://www.pivotaltracker.com/story/show/183613892

**Pull request merged**